src/conn/conn_dhandle.c

```diff
@@ -63,17 +67,31 @@ __conn_dhandle_config_set(WT_SESSION_IMPL *session)
     WT_ERR(__wt_calloc_def(session, 3, &dhandle->cfg));
     switch (dhandle->type) {
     case WT_DHANDLE_TYPE_BTREE:
+        cfg[0] = metaconf;
+        cfg[1] = "checkpoint=()";
+        cfg[2] = NULL;
         WT_ERR(__wt_strdup(session, WT_CONFIG_BASE(session, file_meta), &dhandle->cfg[0]));
+        WT_ASSERT(session, dhandle->meta_base == NULL);
```

Comment:
I'd like a comment here describing what we are saving in meta_base and why. Something like: "Save the metadata string that we know won't change between subsequent checkpoints, to save calculating it repeatedly over the lifetime of the handle"

Reply:
I pushed a large comment. Give it a read. If it's okay I'll merge this branch on Monday since I'm away the whole weekend, I don't want to possibly break the world.